src/classes/UTIL_LightningComponentSupport.cls

```diff
@@ -36,23 +36,43 @@
 
 public with sharing class UTIL_LightningComponentSupport {
 
+    public static final String LIGHTNING_FIELD_TYPE_NUMBER = 'number';
+    public static final String LIGHTNING_FIELD_TYPE_TEXT = 'text';
+    public static final String LIGHTNING_FIELD_TYPE_CHECKBOX = 'checkbox';
+    public static final String LIGHTNING_FIELD_TYPE_DATE = 'date';
+    public static final String LIGHTNING_FIELD_TYPE_DATE_LOCAL = 'date-local';
+
     /*******************************************************************************************************
     * @description Converts Apex field data type to type expected by lightning:datatable and other components
     * @param rawType: the data type from DescribeFieldResult
     * @return the datatype string expected by lightning components
     */
-    public static String getLightningDataType(String rawType) {
-        rawType = rawType.toLowerCase();
-        if (rawType == 'datetime') {
-            return 'date';
-        } else if (rawType == 'date') {
-            return 'date-local';
-        } else if (rawType == 'double') {
-            return 'number';
-        } else if (rawType == 'string') {
-            return 'text';
+    public static String getLightningDataType(Schema.DisplayType fieldDisplayType) {
```

Comment:
Could we also rename this to getLightningDataTableDataType, so we're super clear what this is in the future? We should also update the param in the comment above to reflect the Schema.DisplayType, and the description to reflect only lightning data table.